src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -106,6 +105,7 @@ protected override void Dispose(bool disposing)
             while (PendingRenderBatches.TryDequeue(out var entry))
```

Comment:
The preceding line (which GitHub won't let me comment on) is `base.Dispose(true);`.

For safety, do you think we should consider disposing all the `entry.Data` values *before* calling `base.Dispose`, just in case `base.Dispose` itself throws? Or do a `try`/`finally` here?

Reply:
We don't expect our Dispose APIs (in this case it' `Renderer.Dispose`) to throw. But since it involves clearing shared state (removing from renderRegistry), I guess it doesn't hurt to be defensive.